modules/cli/src/main/scala/scala/cli/commands/Compile.scala

```diff
@@ -51,14 +51,25 @@ object Compile extends ScalaCommand[CompileOptions] {
         if (allowExit)
           sys.exit(1)
       }
-      else if (options.classPath)
-        for {
-          build <- builds.get(Scope.Test).orElse(builds.get(Scope.Main))
-          s     <- build.successfulOpt
-        } {
-          val cp = s.fullClassPath.map(_.toAbsolutePath.toString).mkString(File.pathSeparator)
-          println(cp)
-        }
+      else {
+        if (options.classPath)
+          for {
+            build <- builds.get(Scope.Test).orElse(builds.get(Scope.Main))
+            s     <- build.successfulOpt
+          } {
+            val cp = s.fullClassPath.map(_.toAbsolutePath.toString).mkString(File.pathSeparator)
+            println(cp)
+          }
```

Comment:
Let's factor things a bit
```suggestion
        val successulBuildOpt =
          for {
            build <- builds.get(Scope.Test).orElse(builds.get(Scope.Main))
            s     <- build.successfulOpt
          } yield s
        if (options.classPath)
          for (s <- successulBuildOpt) {
            val cp = s.fullClassPath.map(_.toAbsolutePath.toString).mkString(File.pathSeparator)
            println(cp)
          }
```

Reply:
done as suggested